### PR TITLE
Update package structure for python-project-template

### PR DESCRIPTION
**Pull request by maxrjones**

### CODE REVIEW

File: .pre-commit-config.yaml
Comment (norlandrhagen) [inline .pre-commit-config.yaml:1]: Is the timing change just for less auto-update annoyances?

File: carbonplan_project/__init__.py [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
 from intake import open_catalog
 
+try:
+    __version__ = _version(__name__)
+except _PackageNotFoundError:
+    # package is not installed
+    __version__ = "unknown"
+
+
 cat_dir = pathlib.Path(__file__)
```
Comment (norlandrhagen) [inline carbonplan_project/__init__.py:14]: I haven't looked at the catalog import, but I'm guessing it's probably out of date. Should we remove/comment out this for now or just leave it in?

File: pyproject.toml [hunk 2]
```diff
@@ -1,39 +1,55 @@
+[build-system]
+requires = [
+    "setuptools>=64",
+    "setuptools-scm[toml]>=6.2"
+]
+build-backend = "setuptools.build_meta"
+
 
 [project]
 name = "carbonplan-project"
```
Comment (norlandrhagen) [inline pyproject.toml:10]: Does putting any field that the repo user has to change in <> help? We could also detail all the fields that should be updated in the README.md
ex: name = "<project-name>"

**Review by norlandrhagen — APPROVED** [review]
Looks great!

**Comment by maxrjones** [comment]
Thanks for the review @norlandrhagen! I'm going to address the comments in a separate PR but wanted these merged in before our discussion.

File: .pre-commit-config.yaml
Comment (maxrjones) [reply .pre-commit-config.yaml:1]: yes, based on the suggestions in the other repo update discussion

**Pull request merged**